shared/secrets_masker/tests/secrets_masker/test_secrets_masker.py

```diff
@@ -243,10 +253,15 @@ def test_masking_in_explicit_context_exceptions(self, logger, caplog):
         ],
     )
     def test_mask_secret(self, name, value, expected_mask):
-        filt = SecretsMasker()
-        filt.add_mask(value, name)
+        from airflow_shared.secrets_masker.secrets_masker import _secrets_masker
+        filt = _secrets_masker()
+        configure_secrets_masker_for_test(filt)
 
-        assert filt.patterns == expected_mask
+        try:
+            filt.add_mask(value, name)
+            assert filt.patterns == expected_mask
+        finally:
+            reset_secrets_masker()
```

Comment:
Why do we need to reset anything? Aren't all the changes made on to the `filt` object?

Reply:
Maybe we can  get rid of **global** secrets masker completely and use secrets masker implicitly iniitialized and passed as object down the chain?

Reply:
@potiuk the global singleton was added so that the symlink doesn't cause a lot of chaos and if it does, is much more manageable because we can know that it is a module level global object

Earlier it was looking in the logging filters etc etc.

Reply:
> Why do we need to reset anything? Aren't all the changes made on to the filt object?

We do not, using the `filt` object

Reply:
Moved `should_hide_value_for_key` to SecretsMasker and updated the usages of it too